### ISSUE TITLE
wrong schedulingThroughput in density test

### ISSUE THREAD

**Issue by xial-thu** [kubernetes/perf-tests#1083]
/kind bug
/area clusterloader

When calculating scheduling throughput in density test, I found the calculation is incorrect when running a small test of 300 saturation pods on 16 nodes. The log shows

> I0227 17:38:29.247294 2069840 scheduling_throughput.go:103] SchedulingThroughput: starting collecting throughput data
I0227 17:38:34.247614 2069840 scheduling_throughput.go:118] SchedulingThroughput: labelSelector(group = saturation): 15 pods scheduled
I0227 17:38:39.247969 2069840 scheduling_throughput.go:118] SchedulingThroughput: labelSelector(group = saturation): 127 pods scheduled
I0227 17:38:44.248357 2069840 scheduling_throughput.go:118] SchedulingThroughput: labelSelector(group = saturation): 300 pods scheduled
I0227 17:38:49.248779 2069840 scheduling_throughput.go:118] SchedulingThroughput: labelSelector(group = saturation): 300 pods scheduled
I0227 17:38:54.249241 2069840 scheduling_throughput.go:118] SchedulingThroughput: labelSelector(group = saturation): 300 pods scheduled
I0227 17:38:55.132166 2069840 scheduling_throughput.go:131] SchedulingThroughput: gathering data

300 pods are scheduled within 15 seconds but `gather` stage happens after 25 seconds. In .json output, the average throughput is 12, which is exactly **300/25**. But wait, we should exclude data when all pods are scheduled.

```
{
  "average": 12,
  "perc50": 3,
  "perc90": 34.6,
  "perc99": 34.6
}
```

Below is the part of configuration

```yaml
- name: Collecting saturation pod measurements
  measurements:
  - Identifier: WaitForRunningSaturationDeployments
    Method: WaitForControlledPodsRunning
    Params:
      action: gather
- measurements:
  - Identifier: SaturationPodStartupLatency
    Method: PodStartupLatency
    Params:
      action: gather
- measurements:
  - Identifier: SchedulingThroughput
    Method: SchedulingThroughput
    Params:
      action: gather
      threshold: {{$SCHEDULER_THROUGHPUT_THRESHOLD}}
```

**Comment by wojtek-t** [comment]
This is a know issue - we're going to rewrite this measurement to be based on prometheus metrics.

@mm4tt - FYI

**Comment by mm4tt** [comment]
Yeah, thanks for opening the issue. I talked about this some time ago with @jprzychodzen, he'll be looking into rewriting the measurements in the coming weeks.

**Comment by jprzychodzen** [comment]
@xial-thu thanks for bringing up this issue. Right now I've removed this confusing metric.

We are not using it in any test, and as stated already, there are know issues with this metric. Unless there is an exact use case for it, it's hard to discuss what should this metric mean.

Closing as resolved.

**Comment by xial-thu** [comment]
@jprzychodzen thx for your reply. I remember 50 pod/s is set as a goal in community's document. So I'm a little surprised and confusing when you said you never use it. Would you explain a bit more?

It's not an exact use case for me. I'm just trying to set up a baseline for my system and evaluating possible key metrics. From my point of view, the official SLOs involves too many components and once they're not met, it's hard to debug the bottleneck. I agree that throughput is ambiguous and if we consider it as 'algorithms + binding', creating a pod is much faster than really starting it. If in most cases it doesn't make sense, it's okay to me to ignore it.

Looking forward to your reply.

**Comment by jprzychodzen** [comment]
@xial-thu thanks for you response.

Just to clarify - we didn't use average value of scheduling throughput metric. However, we are using maximum value (a little has changed with  #1093  form 99th percentile). In result, we were providing confusing metric (avg of scheduling throughput) that we did not use at all.

I've closed this issue as it was explicitly targeting avg of scheduling throughput - a metric that was just removed. To point to the right place in a code - we are targeting a maximum value for some X second window: [scheduling_throughput.go](https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/common/scheduling_throughput.go#L150). We are still using this metric and we are measuring it.

Hope that this helps.
